lib/ddtrace/contrib/rails/core_extensions.rb

```diff
@@ -163,49 +163,36 @@ def patch_process_action
         else
           # Rewrite module that gets composed into the Rails controller base class
           ::ActionController::Instrumentation.class_eval do
-            def process_action_with_datadog(*args)
```

Comment:
I think it's great that we're adding tests to cover a scenario we didn't test previously. But I don't think it's a good time to make big changes to how this works. The implementation might be a bit ugly, but the added benefit of making it a little cleaner vs the risk of introducing bugs is not a good trade. In the longer term, we'd like to drop support for 1.9 altogether, so we'll just simply remove this code instead of refactoring it.

Reply:
I don't see it as big change. Code deduplication here is used to make sure there are no differences in behavior when running on ruby  1.9.3 or 2.00. 

I touched this code, because before I found another way, I thought that I need to add some additional behavior to both methods. Instead of duplicating code I decided to ensure both implementations will allways be the same. But since that went away, I still left the change as in my opinion it improves the existing code. 

AFAIK there is no timeline for deprecating 1.9.3 support so at this moment, I think its still worthwhile to treat 1.9.3 as first class citizen.